core/src/main/scala/tech/beshu/ror/api/TestSettingsApi.scala

```diff
@@ -0,0 +1,260 @@
+import scala.language.postfixOps
+import scala.util.Try
+
+class TestSettingsApi(rorInstance: RorInstance)
+                     (implicit clock: Clock) {
+
+  def call(request: TestSettingsRequest)
+          (implicit requestId: RequestId): Task[TestSettingsResponse] = {
+    val testSettingsResponse = request.aType match {
+      case Type.ProvideTestSettings => loadCurrentTestSettings()
+      case Type.UpdateTestSettings => updateTestSettings(request.body)
+      case Type.InvalidateTestSettings => invalidateTestSettings()
+      case Type.ProvideLocalUsers => provideLocalUsers()
+    }
+    testSettingsResponse
+      .executeOn(RorSchedulers.restApiScheduler)
+  }
+
+  private def updateTestSettings(body: String)
+                                (implicit requestId: RequestId): Task[TestSettingsResponse] = {
+    val result = for {
+      config <- rorConfigAndTtlFrom(body)
+      _ <- forceReloadTestSettings(config._1, config._2)
+    } yield ()
+    result.value.map {
+      case Right(_) => TestSettingsResponse.UpdateTestSettings.SuccessResponse("updated settings")
+      case Left(failure) => failure
+    }
+  }
+
+  private def invalidateTestSettings()
+                                    (implicit requestId: RequestId): Task[TestSettingsResponse] = {
+    rorInstance
+      .invalidateImpersonationEngine()
+      .map { _ =>
+        TestSettingsResponse.InvalidateTestSettings.SuccessResponse("ROR Test settings are invalidated")
+      }
+  }
+
+  private def rorConfigAndTtlFrom(payload: String): EitherT[Task, UpdateTestSettings.FailedResponse, (RawRorConfig, FiniteDuration)] = {
+    for {
+      json <- EitherT.fromEither[Task](io.circe.parser.parse(payload).left.map(_ => TestSettingsResponse.UpdateTestSettings.FailedResponse("JSON body malformed")))
+      ttl <- EitherT.fromEither[Task](ttlValue(json))
+      rorConfig <- settingsValue(json)
+    } yield (rorConfig, ttl)
+  }
+
+  private def settingsValue(json: io.circe.Json) = {
+    def liftFailure(failureMessage: String) = EitherT.leftT[Task, RawRorConfig](TestSettingsResponse.UpdateTestSettings.FailedResponse(failureMessage))
+
+    json \\ "settings" match {
+      case Nil =>
+        liftFailure("Malformed settings payload - no settings key")
+      case configJsonValue :: Nil =>
+        configJsonValue.asString match {
+          case Some(configString) =>
+            EitherT(RawRorConfig.fromString(configString).map(_.left.map(error => TestSettingsResponse.UpdateTestSettings.FailedResponse(error.show))))
+          case None =>
+            liftFailure("Malformed settings payload - settings key value is not string")
+        }
+      case _ =>
+        liftFailure("Malformed settings payload - only one settings value allowed")
+    }
+  }
+
+  private def ttlValue(json: io.circe.Json) = {
+    def liftFailure(failureMessage: String) = Left(TestSettingsResponse.UpdateTestSettings.FailedResponse(failureMessage))
+
+    json \\ "ttl" match {
```

Comment:
the extraction of value under a specified key in this method is similar to the one above. And probably to the ConfigApi method. Maybe we can make util from these?

Or maybe we should refactor both API classes and use Circe decoders?